### DISCUSSION TITLE
[PSD2] Esercizi Slide n. 5

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#129] [Esercizi (A-L)]
1. Convertire i seguenti numeri decimali in complemento a 2 utilizzando 8 bits. Dividere il risultato per 4 utilizzando lo shift **a destra**. Riconvertire i numeri in decimale.
	
	(a) 108
	(b) -13
	(c) 105
	(d) -128
	(e) -99

2. Convertire i seguenti numeri decimali in complemento a 2 utilizzando 8 bits. Moltiplicare il risultato per 4 utilizzando lo shift **a sinistra**. Riconvertire i numeri in decimale.
	
	(a) 108
	(b) 25
	(c) -33
	(d) -32

**Comment by Elia-Belli** [comment]
- **Decimale a Ca2 e Dividere per 4 (8bit)**
(a) $108_{10} = 01101100>>2 = 00011011 = 27_{10}$
(b) $-13_{10} = 11110011>>2 = 11111100 = -4_{10}$
(c) $105_{10} = 01101001>>2 = 00011010 = 26_{10}$
(d) $-128_{10} = 10000000 >> 2 = 11100000 = -32_{10}$
(e) $-99_{10} = 10011101 >> 2 = 11100111 = -25_{10}$

- **Decimale a Ca2 e Moltiplicare per 4 (8bit)**
(a) $108_{10} = 01101100<<2 = 10110000$ (overflow)
(b) $25_{10} = 00011001 << 2 = 01100100 = 100_{10}$
(c) $-33_{10} = 11011111 << 2 = 01111100$ (overflow)
(d) $-32_{10} = 11100000 << 2= 10000000 = -128_{10}$
